src/commands/up.rs

```diff
@@ -261,6 +268,85 @@ impl UpCommand {
             )
         })
     }
+
+    // Prepares the application for trigger execution returning the trigger command
+    // to execute and the URL of the locked application.
+    async fn prepare_locked_app(
+        &self,
+        mut locked_app: LockedApp,
+        working_dir: &Path,
+    ) -> Result<(Vec<String>, String)> {
+        // Apply --env to component environments
+        if !self.env.is_empty() {
+            for component in locked_app.components.iter_mut() {
+                component.env.extend(self.env.iter().cloned());
+            }
+        }
+
+        let trigger_command = trigger_command_from_locked_app(&locked_app)?;
+        let locked_app_url = self.write_locked_app(&locked_app, working_dir).await?;
+        Ok((trigger_command, locked_app_url))
+    }
+
+    async fn prepare_app_from_oci(
+        &self,
+        reference: &str,
+        working_dir: &Path,
+    ) -> Result<(Vec<String>, String)> {
+        let mut client = spin_publish::oci::client::Client::new(self.insecure, None)
```

Comment:
Oh dear, I hadn't clocked that Client lived in `spin_publish`, which feels off when we're in a "pull" situation.  Not an issue for this PR though!